streams/src/main/java/org/apache/kafka/streams/processor/internals/SinkNode.java

```diff
@@ -69,7 +70,18 @@ public void init(ProcessorContext context) {
     @Override
     public void process(final K key, final V value) {
         RecordCollector collector = ((RecordCollector.Supplier) context).recordCollector();
-        collector.send(new ProducerRecord<>(topic, null, context.timestamp(), key, value), keySerializer, valSerializer, partitioner);
+
+        final ProducerRecord record;
+        try {
+            record = new ProducerRecord<>(topic, null, context.timestamp(), key, value);
+        } catch (final IllegalArgumentException e) {
+            if (e.getMessage().startsWith("Invalid timestamp:")) {
+                throw new StreamsException("Input topic record has invalid timestamp. Use a different TimestampExtractor to process this data.", e);
```

Comment:
Maybe we can also add sth. like `possibly because an older versioned client is used to send input topic messages to Kafka that do not have timestamps encoded` ..